### PR TITLE
OHARA-1351 Add "cluster name" to Configurator APIs

### PR DESCRIPTION
**Pull request by chia7712**

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/Parameters.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Objects
+
+/**
+  * a collection of query parameter
+  */
+object Parameters {
+  val CLUSTER_NAME: String = "cluster"
+
+  /**
+    * CLUSTER is our first query parameter. We introduce this method do append cluster parameter to url.
```
Comment (konekoya) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/Parameters.scala]: to?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ClusterInfo.scala [hunk 2]
```diff
@@ -16,6 +16,10 @@
 
 package com.island.ohara.client.configurator.v0
 
+/**
+  * There are many kinds of cluster hosted by ohara. We extract a interface to define "what" information should be included by a "cluster
```
Comment (konekoya) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ClusterInfo.scala]: an interface :)

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/AccessWithCluster.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package com.island.ohara.client.configurator.v0
+
+import spray.json.RootJsonFormat
+
+import scala.concurrent.Future
+
+/**
+  * Some APIs can specify target cluster so this class add a variant "add" to carry the target cluster name.
```
Comment (konekoya) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/AccessWithCluster.scala]: adds

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/CollieUtils.scala [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import com.island.ohara.client.configurator.v0.BrokerApi.BrokerClusterInfo
+import com.island.ohara.client.configurator.v0.WorkerApi.WorkerClusterInfo
+import com.island.ohara.client.kafka.WorkerClient
+import com.island.ohara.kafka.BrokerClient
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+  * TODO: this is just a workaround in ohara 0.2. It handle the following trouble:
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/CollieUtils.scala]: handles :)

File: ohara-configurator/src/main/scala/com/island/ohara/configurator/route/NodesRoute.scala [hunk 5]
```diff
@@ -21,78 +21,96 @@ import com.island.ohara.agent.ClusterCollie
...
+          .clusters()
+          .map(_.filter(_._1.nodeNames.contains(res.name)).map(_._1.name).toSeq)
+          .map { wks =>
+            (zks, bks, wks)
+          }
+    }
+    .recover {
+      // all exceptions are swallowed since the ssh info may be wrong.
+      case e: Throwable =>
+        LOG.error("there is a invalid node!!!", e)
```
Comment (konekoya) [inline ohara-configurator/src/main/scala/com/island/ohara/configurator/route/NodesRoute.scala]: an

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
@konekoya Thanks for reviews!!!

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
retry connector

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry common

**Comment by chia7712** [comment]
retry

**Comment by chia7712** [comment]
failed test (TestK8SSimple.testK8SClientlog) is traced by OHARA-1391

**Comment by chia7712** [comment]
retry

**Pull request merged**